test/EndToEndSpec.hs

```diff
@@ -638,6 +638,39 @@ spec = around_ (inTempDirectoryNamed "my-package") $ do
           - "*.markdown"
         |] `shouldWarn` ["Specified pattern \"*.markdown\" for extra-doc-files does not match any files"]
 
+    describe "extra-files" $ do
+      it "accepts a list of files" $ do
+        touch "CHANGES.markdown"
+        touch "README.markdown"
+        [i|
+        extra-files:
+          - CHANGES.markdown
+          - README.markdown
+        |] `shouldRenderTo` (package [i|
+        extra-files:
+            CHANGES.markdown
+            README.markdown
+        |]) {packageCabalVersion = "3.14"}
+
+      it "accepts glob patterns" $ do
+        touch "CHANGES.markdown"
+        touch "README.markdown"
+        [i|
+        extra-files:
+          - "*.markdown"
+        |] `shouldRenderTo` (package [i|
+        extra-files:
+            CHANGES.markdown
+            README.markdown
+        |]) {packageCabalVersion = "3.14"}
+
+      it "warns if a glob pattern does not match anything" $ do
+        [i|
+        name: foo
+        extra-files:
+          - "*.markdown"
+        |] `shouldWarn` ["Specified pattern \"*.markdown\" for extra-files does not match any files"]
```

Comment:
We only need one basic test case here.

Rational:

When you do TDD, you don't test everything exhaustively.  You only write as many tests as you need to drive the implementation. In this specific case we reuse a lot of existing code, which is already covered by other tests.  

Technically we  could add 10, or a 100, or a 1000 `extra-whatever-files`, reusing the existing code.  That does not mean that we would need to add x3 test cases for those.

One test case is enough here.

```haskell
describe "extra-files" $ do
  it "accepts extra-files" $ do
    ...
```

Reply:
Now, there is this potential risk that `extra-doc-files` could be removed in the future, and then we lose test coverage for `extra-files`.

If you wanna guard against this, then what you want to do is make those three test cases reusable and then reuse the same test code to test those fields.

If that's the approach you wanna take, then please feel free to give it a try.

Reply:
Happy to cut it down to one test. I can't see `extra-doc-files` being removed from package descriptions any time soon (there is no overlap between `extra-doc-files` and `extra-files`).